### ISSUE TITLE
Xamarin iOS Firebase App.Configure breaks deep links

### ISSUE THREAD

**Issue by justintoth** [xamarin/GoogleApisForiOSComponents#279]
In a Xamarin iOS app, I have configured deep (universal) links to work. If a deep link is clicked, it properly calls AppDelegate.ContinueUserActivity, where I can parse the url and navigate the user somewhere within my app. However, if you integrate in the Firebase SDK and call:

`Firebase.Core.App.Configure ();`

...from within the AppDelegate.FinishedLaunching method (which is the recommended approach) or from anywhere else in code, then when clicking on a deep link it still launches the app but it no longer calls AppDelegate.ContinueUserActivity so no action can take place based on the deep link url.

It's as if Firebase is taking over handling of deep links, so we can no longer do so ourselves.

**Comment by justintoth** [comment]
Can someone investigate this? We're going to have to revert back from Firebase to Google Analytics in the short term, because deep links are critical to our app. And using GA is not desirable, as it's going away in October.

**Comment by tele-bird** [comment]
Hi, @justintoth.  I'm not part of the triage team, but I'd suggest adding this line of code right after your call to App.Configure():

Firebase.DynamicLinks.DynamicLinks.PerformDiagnostics(null);

It'll write some diagnostics to Output, and you might find your problem there.

**Comment by justintoth** [comment]
@tele-bird Nothing useful that I can see.

> ---- Firebase Dynamic Links diagnostic output start ----
> Firebase Dynamic Links framework version 3.0.1
> System information: OS iOS, OS version 12.3.1, model iPhone
> Current date 2019-06-27 16:51:55 +0000
> Device locale en-US (raw en_US), timezone America/New_York
> 	Specified custom URL scheme is org.narrpr.mobile.dev and Info.plist contains such scheme in CFBundleURLTypes key.
> 	AppID Prefix: 75XPPVJ3JR, Team ID: 75XPPVJ3JR, AppId Prefix equal to Team ID: YES
> performDiagnostic completed successfully! No errors found.
> ---- Firebase Dynamic Links diagnostic output end ----

**Comment by tele-bird** [comment]
@justintoth , yep, looks good to me too!

Actually, I just realized your problem...  Whether ContinueUserActivity is fired depends on your Universal Links configuration.  If it's not fired, then it's definitely a problem with Universal Links (and not Firebase).  I find this article helpful for understanding: https://blog.branch.io/ios-9-2-deep-linking-guide-transitioning-to-universal-links/

**Comment by justintoth** [comment]
@tele-bird Sorry I don't buy it. ContinueUserActivity is fired correctly if Firebase.Core.App.Configure isn't called, which indicates that Universal Links are working correctly (and they have worked correct for years.) Only after calling Firebase.Core.App.Configure does it break and ContinueUserActivity isn't called. I can even call Firebase.Core.App.Configure WITHIN ContinueUserActivity, and then the first deep link will work and then the second will not.

**Comment by SotoiGhost** [comment]
Is this still an issue?

**Comment by SotoiGhost** [comment]
We have not received the requested information. If you are still experiencing this issue, please, open a new issue related to this. Thanks!

**Comment by justintoth** [comment]
Yes it's resolved. For others who run into the same issue the resolution is to add a new key to your Info.plist file:

Key: FirebaseAppDelegateProxyEnabled Type: Boolean Value: No (false)

After that Firebase stops hijacking handling of deep link urls, so AppDelegate.ContinueUserActivity is correctly called when a deep link is clicked.
